### PR TITLE
legend perf improvement

### LINKED ISSUE #3089: calculating legend layout during draw is extremely slow
Calling redraw on a legend with 431 items locks the browser for 9 seconds. Culprit seems to be [Legend._calculateLayoutInfo](https://github.com/palantir/plottable/blob/develop/src/components/legend.ts#L162-L191)

screen shot 2016-07-23 at 5 44 34 pm

Seems like an easy performance win to speed up draw times. Especially important since the responsive strategy for plottable is to redraw on resize events!

### PR DESCRIPTION
**Pull request by hellochar**
Just use CacheMeasurer everywhere

Fixes #3089

### CODE REVIEW

**Comment by themadcreator** [comment]
I think I had originally gone through and replaced every since measurer with a cachemeasurer, but some of the plots didn't match after that. How thoroughly did you review the existing dev examples?

**Comment by hellochar** [comment]
Not closely. Do you remember what the issue was?

File: tslint.json [hunk 1]
```diff
@@ -2,6 +2,11 @@
   "rules": {
     "align": [true, "parameters", "statements"],
     "arrow-parens": true,
+    "ban": [
+      true,
+      "eval",
+      { "name": ["Typesettable", "Measurer"], "message": "Use CacheMeasurer instead" }
```
Comment (themadcreator) [inline tslint.json:8]: heh. 👍

**Review by themadcreator — APPROVED** [review]
Cool, LGTM

**Pull request merged**